src/gsy_e/models/strategy/__init__.py

```diff
@@ -511,7 +519,7 @@ def post_offer(self, market, replace_existing=True, **offer_kwargs) -> Offer:
         """
         if replace_existing:
             # Remove all existing offers that are still open in the market
-            self.offers.remove_offer_from_cache_and_market(market)
+            self._remove_existing_offers(market, offer_kwargs.get("time_slot") or market.time_slot)
```

Comment:
Can you please give a bid of context here? AFAIS: `self.offers.remove_offer_from_cache_and_market` also dealt with deleting all offers from the market (if `offer_id` was set to None, which was done here before).
So, either it is able to delete all orders of one asset from the market or we need to remove this functionality from this method. 
Curious about your explanation.

Reply:
I did this to avoid deletion of all bids/offers when somebody posts a new bid/offer on the future market (and sets replace_existing=True). IMO, this is a bug and is evident in existing unit tests that I have fixed. What do you think?

Reply:
> I did this to avoid deletion of all bids/offers when somebody posts a new bid/offer on the future market (and sets replace_existing=True

We are talking about bids/offers of the same asset right? AFAIS, `self.offers.remove_offer_from_cache_and_market` already took care of this. 

> is evident in existing unit tests that I have fixed 

can you please point me to it? Thanks!

Reply:
Sure. Lines 147-151 in [here](https://github.com/gridsingularity/gsy-e/pull/1566/files#diff-e5331d4b9154206362fa92d2e6f72dd3003a688e680626f0ec535f65af9a79f6L147). This test was posting offers for different timeslots in the future but only the last one remained because it used to flush the offers cache before posting a new one!

Reply:
Ahh, now I get it. The `post_offer(..., replace_existing=True)` was not respecting the market slot and was deleting all orders from all future markets before. Now it reads the `market_slot` and calls `_remove_existing_offers` where `get_posted_offers` finds the list of offers that are posted in the future market. Right?
(Sorry for thinking out loud)